diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59d6a8c2e..119da862a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,16 +73,6 @@ pkg_get_variable(gettext_domain     gldi plugins_gettext)
 pkg_get_variable(gtkversion         gldi gtkversion)
 pkg_get_variable(pkgdatadir         gldi pkgdatadir)
 
-# check that installation dir matches with the core
-if (NOT "${CMAKE_INSTALL_PREFIX}" STREQUAL "${prefix}")
-	if (NOT CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-		message (STATUS "It seems that the current CMAKE_INSTALL_{PREFIX,LIBDIR,DATAROOTDIR} flags are not set to what is expected by the core.")
-		message (STATUS " It will be replaced by this value: ${prefix}")
-		message (WARNING "Plug-ins should be installed in the directory defined when setting up the core, that is to say in ${pluginsdir}")
-	endif()
-	set (CMAKE_INSTALL_PREFIX "${prefix}")
-endif()
-
 # set internationalisation
 set (GETTEXT_PLUGINS "${gettext_domain}")
 # set base data directories
